README.md
# repository.worldtvrepo-1.6


changelog.txt
[B]Version 5.0[/B]
- Initial version


program.worldtvclean/default.py

import urllib,urllib2,re,uuid, time
import xbmcgui,xbmcplugin
import os


thumbnailPath = xbmc.translatePath('special://thumbnails');
cachePath = os.path.join(xbmc.translatePath('special://home'), 'cache')
tempPath = xbmc.translatePath('special://temp')
addonPath = os.path.join(os.path.join(xbmc.translatePath('special://home'), 'addons'),'program.RockClean')
mediaPath = os.path.join(addonPath, 'media')
databasePath = xbmc.translatePath('special://database')

class cacheEntry:
    def __init__(self, namei, pathi):
        self.name = namei
        self.path = pathi
    
def mainMenu():
    xbmc.executebuiltin("Container.SetViewMode(500)")
    addItem('[COLOR red]Clear Cache[/COLOR]','url', 1,os.path.join(mediaPath, "cache.png"))
    addItem('[COLOR red]Delete Thumbnails[/COLOR]', 'url', 2,os.path.join(mediaPath, "thumbs.png"))
    addItem('[COLOR red]Purge Packages[/COLOR]', 'url', 3,os.path.join(mediaPath, "packages.png"))
    addItem('[COLOR red]Update WorldTV[/COLOR]', 'url', 4,os.path.join(mediaPath, "update.png"))

def addLink(name,url,iconimage):
	ok=True
	liz=xbmcgui.ListItem(name, iconImage="DefaultVideo.png", thumbnailImage=iconimage)
	liz.setInfo( type="Video", infoLabels={ "Title": name } )
	ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=url,listitem=liz)
	return ok


def addDir(name,url,mode,iconimage):
	u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)
	ok=True
	liz=xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
	liz.setInfo( type="Video", infoLabels={ "Title": name } )
	ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=True)
	return ok
	
def addItem(name,url,mode,iconimage):
	u=sys.argv[0]+"?url="+urllib.quote_plus(url)+"&mode="+str(mode)+"&name="+urllib.quote_plus(name)
	ok=True
	liz=xbmcgui.ListItem(name, iconImage="DefaultFolder.png", thumbnailImage=iconimage)
	liz.setInfo( type="Video", infoLabels={ "Title": name } )
	ok=xbmcplugin.addDirectoryItem(handle=int(sys.argv[1]),url=u,listitem=liz,isFolder=False)
	return ok
      
def get_params():
	param=[]
	paramstring=sys.argv[2]
	if len(paramstring)>=2:
			params=sys.argv[2]
			cleanedparams=params.replace('?','')
			if (params[len(params)-1]=='/'):
					params=params[0:len(params)-2]
			pairsofparams=cleanedparams.split('&')
			param={}
			for i in range(len(pairsofparams)):
					splitparams={}
					splitparams=pairsofparams[i].split('=')
					if (len(splitparams))==2:
							param[splitparams[0]]=splitparams[1]
							
	return param   
	
def setupCacheEntries():
    entries = 5 #make sure this refelcts the amount of entries you have
    dialogName = ["WTF", "4oD", "BBC iPlayer", "Simple Downloader", "ITV"]
    pathName = ["special://profile/addon_data/plugin.video.whatthefurk/cache", "special://profile/addon_data/plugin.video.4od/cache",
					"special://profile/addon_data/plugin.video.iplayer/iplayer_http_cache","special://profile/addon_data/script.module.simple.downloader",
                    "special://profile/addon_data/plugin.video.itv/Images"]
                    
    cacheEntries = []
    
    for x in range(entries):
        cacheEntries.append(cacheEntry(dialogName[x],pathName[x]))
    
    return cacheEntries


def clearCache():
    
    if os.path.exists(cachePath)==True:    
        for root, dirs, files in os.walk(cachePath):
            file_count = 0
            file_count += len(files)
            if file_count > 0:

                dialog = xbmcgui.Dialog()
                if dialog.yesno("[COLOR limegreen]Delete WorldTV Cache Files[/COLOR]", str(file_count) + "[COLOR  limegreen] files found[/COLOR]", "[COLOR gold]Do you want to delete them?[/COLOR]"):
                
                    for f in files:
                        try:
                            if (f == "xbmc.log" or f == "xbmc.old.log"): continue
                            os.unlink(os.path.join(root, f))
                        except:
                            pass
                    for d in dirs:
                        try:
                            shutil.rmtree(os.path.join(root, d))
                        except:
                            pass
                        
            else:
                pass
    if os.path.exists(tempPath)==True:    
        for root, dirs, files in os.walk(tempPath):
            file_count = 0
            file_count += len(files)
            if file_count > 0:
                dialog = xbmcgui.Dialog()
                if dialog.yesno("[COLOR  limegreen]Delete WorldTV Temp Files[/COLOR]", str(file_count) + "[COLOR  limegreen] files found[/COLOR]", "[COLOR gold]Do you want to delete them?[/COLOR]"):
                    for f in files:
                        try:
                            if (f == "xbmc.log" or f == "xbmc.old.log"): continue
                            os.unlink(os.path.join(root, f))
                        except:
                            pass
                    for d in dirs:
                        try:
                            shutil.rmtree(os.path.join(root, d))
                        except:
                            pass
                        
            else:
                pass
    if xbmc.getCondVisibility('system.platform.ATV2'):
        atv2_cache_a = os.path.join('/private/var/mobile/Library/Caches/AppleTV/Video/', 'Other')
        
        for root, dirs, files in os.walk(atv2_cache_a):
            file_count = 0
            file_count += len(files)
        
            if file_count > 0:

                dialog = xbmcgui.Dialog()
                if dialog.yesno("[COLOR limegreen]Delete ATV2 Cache Files[/COLOR]", str(file_count) + "[COLOR gold] files found in 'Other'", "Do you want to delete them?[/COLOR]"):
                
                    for f in files:
                        os.unlink(os.path.join(root, f))
                    for d in dirs:
                        shutil.rmtree(os.path.join(root, d))
                        
            else:
                pass
        atv2_cache_b = os.path.join('/private/var/mobile/Library/Caches/AppleTV/Video/', 'LocalAndRental')
        
        for root, dirs, files in os.walk(atv2_cache_b):
            file_count = 0
            file_count += len(files)
        
            if file_count > 0:

                dialog = xbmcgui.Dialog()
                if dialog.yesno("[COLOR gold]Delete ATV2 Cache Files[/COLOR]", str(file_count) + "[COLOR gold] files found in 'LocalAndRental'[/COLOR]", "[COLOR gold]Do you want to delete them?[/COLOR]"):
                
                    for f in files:
                        os.unlink(os.path.join(root, f))
                    for d in dirs:
                        shutil.rmtree(os.path.join(root, d))
                        
            else:
                pass    
                
    cacheEntries = setupCacheEntries()
                                         
    for entry in cacheEntries:
        clear_cache_path = xbmc.translatePath(entry.path)
        if os.path.exists(clear_cache_path)==True:    
            for root, dirs, files in os.walk(clear_cache_path):
                file_count = 0
                file_count += len(files)
                if file_count > 0:

                    dialog = xbmcgui.Dialog()
                    if dialog.yesno("[COLOR  limegreen]ROCK CLEANER[/COLOR]",str(file_count) + "[COLOR  limegreen]%s [COLOR gold]cache files found[/COLOR]"%(entry.name), "[COLOR gold]Do you want to delete them?[/COLOR]"):
                        for f in files:
                            os.unlink(os.path.join(root, f))
                        for d in dirs:
                            shutil.rmtree(os.path.join(root, d))
                            
                else:
                    pass
                

    dialog = xbmcgui.Dialog()
    dialog.ok("[COLOR  limegreen]WorldTV Cleaner[/COLOR]", "[COLOR gold]Done Clearing All Cache files[/COLOR]")
    
    
def deleteThumbnails():
    
    if os.path.exists(thumbnailPath)==True:  
            dialog = xbmcgui.Dialog()
            if dialog.yesno("[COLOR red]Delete Thumbnails[/COLOR]", "[COLOR gold]This option deletes all WorldTV thumbnails[/COLOR]", "[COLOR gold]Are you sure you want to do this?[/COLOR]"):
                for root, dirs, files in os.walk(thumbnailPath):
                    file_count = 0
                    file_count += len(files)
                    if file_count > 0:                
                        for f in files:
                            try:
                                os.unlink(os.path.join(root, f))
                            except:
                                pass                
    else:
        pass
    
    text13 = os.path.join(databasePath,"Textures13.db")
    os.unlink(text13)
        
    dialog.ok("[COLOR  limegreen]Restart WorldTV[/COLOR]", "[COLOR gold]Please restart WorldTV to rebuild thumbnail library[/COLOR]")
        
def purgePackages():
    
    purgePath = xbmc.translatePath('special://home/addons/packages')
    dialog = xbmcgui.Dialog()
    for root, dirs, files in os.walk(purgePath):
            file_count = 0
            file_count += len(files)
    if dialog.yesno("[COLOR  limegreen]Delete Package Cache Files[/COLOR]", "[COLOR  limegreen]%d [COLOR gold]packages found.[/COLOR]"%file_count, "[COLOR gold]Do you want to Delete Them?[/COLOR]"):  
        for root, dirs, files in os.walk(purgePath):
            file_count = 0
            file_count += len(files)
            if file_count > 0:            
                for f in files:
                    os.unlink(os.path.join(root, f))
                for d in dirs:
                    shutil.rmtree(os.path.join(root, d))
                dialog = xbmcgui.Dialog()
                dialog.ok("[COLOR limegreen]WorldTV CLEANER[/COLOR]", "[COLOR gold]Deleting Packages all done[/COLOR]")
            else:
                dialog = xbmcgui.Dialog()
                dialog.ok("[COLOR blue]WorldTV CLEANER[/COLOR]", "[COLOR gold]No Packages to Purge[/COLOR]") 
                
def ForceUpdate():
  dialog = xbmcgui.Dialog()
  if dialog.yesno("[COLOR limegreen]WorldTV CLEANER[/COLOR]", "[COLOR gold]Update All Add-ons[/COLOR]", "[COLOR gold]And Repository ?[/COLOR]"):
     xbmc.executebuiltin('UpdateAddonRepos()')
     xbmc.executebuiltin('UpdateLocalAddons()') 
  dialog = xbmcgui.Dialog()
  dialog.ok("[COLOR limegreen]WorldTV CLEANER[/COLOR]", "[COLOR gold]Updating All Addons and Repos[/COLOR]")  
                
params=get_params()
url=None
name=None
mode=None

try:
        url=urllib.unquote_plus(params["url"])
except:
        pass
try:
        name=urllib.unquote_plus(params["name"])
except:
        pass
try:
        mode=int(params["mode"])
except:
        pass

if mode==None or url==None or len(url)<1:
        mainMenu()
       
elif mode==1:
		clearCache()
        
elif mode==2:
        deleteThumbnails()

elif mode==3:
		purgePackages()
		
elif mode==4:
		ForceUpdate()



xbmcplugin.endOfDirectory(int(sys.argv[1]))



plugin.video.worldtvwizard-1.0.0/common.py
import os,sys,xbmc,xbmcplugin,xbmcaddon,xbmcgui,urllib,urllib2,re,time,datetime,string,StringIO,logging,random,array,htmllib,xbmcvfs
#import common
addon_id='plugin.program.worldtvmessages'
try: 				from addon.common.addon import Addon
except:
    try: 		from t0mm0.common.addon import Addon
    except: from t0mm0_common_addon import Addon
try: addon_handle=int(sys.argv[1])
except: addon_handle=0
try: addon=Addon(addon_id,sys.argv)
except: addon=Addon(addon_id,0)
try: 				from addon.common.net import Net
except:
    try: 		from t0mm0.common.net import Net
    except: from t0mm0_common_net import Net
net=Net()
settings=xbmcaddon.Addon(id=addon_id)
## ################################################## ##
## ################################################## ##
def gAI(t):
	try: return settings.getAddonInfo(t)
	except: return ""
def show_settings(self): settings.openSettings()
def eod(): xbmcplugin.endOfDirectory(addon_handle)
## ################################################## ##
## ################################################## ##
icon=gAI('icon'); 
fanart=gAI('fanart'); 
addon_name=gAI('name'); 
addon_path=gAI('path'); 
addon_type=gAI('type'); 
addon_author=gAI('author'); 
addon_version=gAI('version'); 
addon_stars=gAI('stars'); 
addon_changelog=gAI('changelog'); 
disclaimer=gAI('disclaimer'); 
description=gAI('description'); 
summary=gAI('summary'); 	
artPath=xbmc.translatePath(os.path.join(addon_path,'art'))
## ################################################## ##
## ################################################## ##
def note(title='',msg='',delay=5000,image=''):
	if len(image)==0: image+=icon
	xbmc.executebuiltin('XBMC.Notification("%s","%s",%d,"%s")' % (title,msg,delay,image))
def SettingG(setting):
	try: return settings.getSetting(setting)
	except: return ""
def SettingS(setting,value):
	settings.setSetting(id=setting,value=value)
def deb(s,t): ### for Writing Debug Data to log file ###
	#if (_debugging==True): 
	print s+':  '+t
def debob(t): ### for Writing Debug Object to log file ###
	#if (_debugging==True): 
	print t
def nolines(t):
	it=t.splitlines(); t=''
	for L in it: t=t+L
	t=((t.replace("\r","")).replace("\n",""))
	return t
def art(f,fe=''): 
	fe1='.png'; fe2='.jpg'; fe3='.gif'; fe4='.wav'; fe5='.txt'; 
	if   fe1 in f: f=f.replace(fe1,''); fe=fe1; 
	elif fe2 in f: f=f.replace(fe2,''); fe=fe2; 
	elif fe3 in f: f=f.replace(fe3,''); fe=fe3; 
	elif fe4 in f: f=f.replace(fe4,''); fe=fe4; 
	elif fe5 in f: f=f.replace(fe5,''); fe=fe5; 
	return xbmc.translatePath(os.path.join(artPath,f+fe))
def artp(f,fe='.png'): return art(f,fe)
def artj(f,fe='.jpg'): return art(f,fe)
def addonPath(f,fe=''): return xbmc.translatePath(os.path.join(addon_path,f+fe))
def cFL(t,c='tan'): return '[COLOR '+c+']'+t+'[/COLOR]' ### For Coloring Text ###
def popYN(title='',line1='',line2='',line3='',n='',y=''):
	diag=xbmcgui.Dialog()
	r=diag.yesno(title,line1,line2,line3,n,y)
	if r: return r
	else: return False
	#del diag
def popOK(msg="",title="",line2="",line3=""):
	dialog=xbmcgui.Dialog()
	#ok=dialog.ok(title, msg, line2, line3)
	dialog.ok(title, msg, line2, line3)
def OPEN_URL(url):
  req=urllib2.Request(url)
  req.add_header('User-Agent','Mozilla/5.0 (Windows; U; Windows NT 5.1; en-GB; rv:1.9.0.3) Gecko/2008092417 Firefox/3.0.3')
  response=urllib2.urlopen(req)
  link=response.read()
  response.close()
  return link
def File_Save(path,data):
    file=open(path,'w')
    file.write(data)
    file.close()
def File_Open(path):
    if os.path.isfile(path): ## File found.
        file=open(path, 'r')
        contents=file.read()
        file.close()
        return contents
    else: return '' ## File not found.
def tfalse(r,d=False): ## Get True / False
	if   (r.lower()=='true' ) or (r.lower()=='t') or (r.lower()=='y') or (r.lower()=='1') or (r.lower()=='yes'): return True
	elif (r.lower()=='false') or (r.lower()=='f') or (r.lower()=='n') or (r.lower()=='0') or (r.lower()=='no'): return False
	else: return d

## ################################################## ##
## ################################################## ##
ACTION_PREVIOUS_MENU 		=  10	## ESC action
ACTION_NAV_BACK 				=  92	## Backspace action
ACTION_MOVE_LEFT 				=   1	## Left arrow key
ACTION_MOVE_RIGHT 			=   2	## Right arrow key
ACTION_MOVE_UP 					=   3	## Up arrow key
ACTION_MOVE_DOWN 				=   4	## Down arrow key
ACTION_MOUSE_WHEEL_UP 	= 104	## Mouse wheel up
ACTION_MOUSE_WHEEL_DOWN = 105	## Mouse wheel down
ACTION_MOUSE_DRAG 			= 106	## Mouse drag
ACTION_MOUSE_MOVE 			= 107	## Mouse move
#
ACTION_SELECT_ITEM			=		7	## ?
ACTION_PARENT_DIR				=		9	## ?
ACTION_CONTEXT_MENU			=	117	## ?
ACTION_NEXT_ITEM				=	 14	## ?
ACTION_BACKSPACE				=	110	## ?
## ################################################## ##
## ################################################## ##
OverlayBorder=artp('blank1'); 
OverlayBackground=artp('ContentPanel'); 
hubLogo=artp('logo-rectangular3'); 
#hubLogo=artp('logo_notice.png'); 
DefaultNoteImage=artp('blank1'); 
DefaultSplitter="|||"; 
DefaultUrl="http://worldtrumprepo.netai.net/Zips/plugin.program.worldtvmessages/wtm.html"; 
class MyWindow(xbmcgui.WindowDialog): #xbmcgui.Window): ##xbmcgui.Window
	scr={}; scr['L']=0; scr['T']=0; scr['W']=1280; scr['H']=720; 
	def __init__(self,noteType='t',noteMessage='',noteImage='',L=140,T=110,W=1000,H=500,Font='font14',TxtColor='0xFF64d1ff'):
		if len(noteImage)==0: noteImage=DefaultNoteImage
		if   (noteType.lower()=='text')  or (noteType.lower()=='t'): noteType='t'
		elif (noteType.lower()=='image') or (noteType.lower()=='i'): noteType='i'
		self.noteType=noteType; self.noteMessage=noteMessage; self.noteImage=noteImage; self.Font=Font; self.TxtColor=TxtColor; 
		## ### ## 
		self.background=OverlayBackground; #artp('black1'); 
		self.BG=xbmcgui.ControlImage(L,T,W,H,self.background,aspectRatio=0,colorDiffuse='0xFF3030FF'); 
		#self.OlayBrdr=xbmcgui.ControlImage(L,T,W,H,OverlayBorder,aspectRatio=0); 
		#self.OlaySplash=xbmcgui.ControlImage(L,T,W,H,icon,aspectRatio=0); 
		iLogoW=144; iLogoH=68; 
		self.iLogo=xbmcgui.ControlImage((L+(W/2))-(iLogoW/2),T+10,iLogoW,iLogoH,hubLogo,aspectRatio=0); 
		## ### ## 
		###L2=L+110; T2=T+130; W2=W-(T2-T)-90; H2=H-(L2-L)-110; #L3=L2+5; T3=T2+5; W3=W2-18; H3=H2-10; 
		##L2=L+87; T2=T+80; W2=W-(T2-T)-96; H2=H-(L2-L)-74; L3=L2+5; T3=T2+60; W3=W2-18; H3=H2-5-60; 
		#L2=L+67; T2=T+60; W2=W-(T2-T)-96; H2=H-(L2-L)-74; 
		L2=200; T2=200; W2=880; H2=340; 
		L3=L2+5; T3=T2+60; W3=W2-18; H3=H2-5-60; 
		self.ImgMessage=xbmcgui.ControlImage(L2,T2,W2,H2,self.noteImage,aspectRatio=0); 
		self.TxtMessage=xbmcgui.ControlTextBox(L2+5,T2,W2-10,H2,font=self.Font,textColor=self.TxtColor); 
		#self.TxtMessage=xbmcgui.ControlTextBox(L3,T3,W3,H3,font=self.Font,textColor=self.TxtColor); 
		#print [self.background,OverlayBorder,self.noteImage]
		## ### ## 
		focus=artp('button-focus_lightblue'); nofocus=artp('button-focus_grey'); 
		w1=120; h1=35; w2=160; h2=35; spacing1=20; 
		l2=L+W-spacing1-w2; t2=T+H-h2-spacing1; 
		l1=L+W-spacing1-w2-spacing1-w1; t1=T+H-h1-spacing1; 
		self.buttonDismiss=xbmcgui.ControlButton(l1,t1,w1,h1,"Thank You ",textColor="0xFF000000",focusedColor="0xFF000000",alignment=2,focusTexture=focus,noFocusTexture=nofocus); 
		self.buttonRemindMe=xbmcgui.ControlButton(l2,t2,w2,h2,"Close Window",textColor="0xFF000000",focusedColor="0xFF000000",alignment=2,focusTexture=focus,noFocusTexture=nofocus); 
		#self.OlaySplash
		for z in [self.BG,self.ImgMessage,self.TxtMessage,self.iLogo,self.buttonRemindMe,self.buttonDismiss]: self.addControl(z); 
		#for z in [self.BG,self.ImgMessage,self.TxtMessage,self.OlayBrdr,self.buttonRemindMe,self.buttonDismiss]: self.addControl(z); 
		#for z in [self.BG,self.OlayBrdr,self.ImgMessage,self.TxtMessage,self.buttonRemindMe,self.buttonDismiss]: self.addControl(z); 
		#self.OlayBrdr.setAnimations([('WindowOpen','effect=fade delay=0 time=0 start=0 end=70')]); 
		#self.ImgMessage.setAnimations([('WindowOpen','effect=fade delay=0 time=0 start=0 end=70')]); 
		#self.OlaySplash.setAnimations([('WindowOpen','effect=fade delay=0 time=8000 start=100 end=0')]); 
		#self.OlaySplash.setAnimations([('WindowOpen','effect=slide delay=0 time=5000 start=-1800 end=1800')]); 
		#self.ImgMessage.setAnimations([('WindowOpen','effect=fade delay=2000 time=2000 start=0 end=100')]); 
		#for z in [self.BG,self.ImgMessage,self.TxtMessage,self.OlayBrdr,self.buttonRemindMe,self.buttonDismiss]: 
		for z in [self.BG,self.ImgMessage,self.TxtMessage,self.iLogo,self.buttonRemindMe,self.buttonDismiss]: 
			#z.setAnimations([('WindowOpen','effect=slide delay=0 time=5000 start=0,-1800 end=0'),('WindowClose','effect=slide delay=0 time=5000 start=0 end=0,-1800')]); 
			z.setAnimations([('WindowOpen','effect=fade delay=0 time=2000 start=0 end=100'),('WindowClose','effect=slide delay=0 time=2000 start=0 end=0,'+str(0-(H+T+10)))]); 
		## ### ## 
		self.buttonRemindMe.controlLeft(self.buttonDismiss); self.buttonRemindMe.controlRight(self.buttonDismiss); 
		self.buttonDismiss.controlLeft(self.buttonRemindMe); self.buttonDismiss.controlRight(self.buttonRemindMe); 
		## ### ## 
		self.TxtMessage.setText(self.noteMessage); 
		self.setFocus(self.buttonRemindMe); 
	def doRemindMeLater(self):
		try:
			SettingS("noteType",self.noteType); 
			SettingS("noteImage",""); 
			SettingS("noteMessage",""); 
		except: pass
		##CODE HERE##
		self.CloseWindow1st()
	def doDismiss(self):
		try:
			SettingS("noteType",self.noteType); 
			SettingS("noteImage",self.noteImage); 
			SettingS("noteMessage",self.noteMessage); 
		except: pass
		##CODE HERE##
		self.CloseWindow1st()
	def onAction(self,action):
		try: F=self.getFocus()
		except: F=False
		if   action == ACTION_PREVIOUS_MENU: self.doRemindMeLater()
		elif action == ACTION_NAV_BACK: self.doRemindMeLater()
		elif action == ACTION_SELECT_ITEM: self.doDismiss()
		else:
			try:
				if not F==self.buttonRemindMe:
					self.setFocus(self.buttonDismiss); 
			except: pass
	def onControl(self,control):
		if   control==self.buttonRemindMe: self.doRemindMeLater()
		elif control== self.buttonDismiss: self.doDismiss()
		else:
			try:
				self.setFocus(self.buttonRemindMe); 
			except: pass
	#def onInit(self): pass
	#def onClick(self,control): pass
	#def onControl(self,control): pass
	#def onFocus(self,control): pass
	#def onAction(self,action): pass
	def CloseWindow1st(self):
		##CODE HERE##
		self.close()
## ################################################## ##
## ################################################## ##
DefaultReturn=["",""]
def FetchNews():
	NewImage=""; NewMessage=""; info_location=addonPath("test.txt"); info_location3=addonPath("url.txt"); 
	info_location2=DefaultUrl; 
	if os.path.isfile(info_location)==True: 
		try: 
			html=File_Open(info_location)
			print info_location
		except: return DefaultReturn
	elif os.path.isfile(info_location3)==True: 
		try: 
			info_location3B=File_Open(info_location3).strip()
			if (info_location3B) > 0: 
				html=OPEN_URL(info_location3B)
				print info_location3B
			else: return DefaultReturn
		except: return DefaultReturn
	else: 
		try: 
			html=OPEN_URL(info_location2)
			print info_location2
		except: return DefaultReturn
	if DefaultSplitter in html:
		NewImage  =html.split(DefaultSplitter)[0].strip(); 
		NewMessage=html.split(DefaultSplitter)[1].strip(); 
	return (NewImage,NewMessage)

def CheckNews(TypeOfMessage,NewImage,NewMessage,DoFromService=True):
	if (len(NewImage) > 0) or (len(NewMessage) > 0):
		debob(["notifications-on-startup",tfalse(SettingG("notifications-on-startup")),"DoFromService",DoFromService])
		if (tfalse(SettingG("notifications-on-startup"))==False) or (DoFromService==False):
			if NewImage.lower()=="none": NewImage=""
			if NewMessage.lower()=="none": NewMessage=""
			OldnoteType=SettingG("noteType"); OldnoteImage=SettingG("noteImage"); OldnoteMessage=SettingG("noteMessage"); 
			OldnoteImage=OldnoteImage.replace(DefaultNoteImage,'')
			if OldnoteImage.lower()=="none": OldnoteImage=""
			if OldnoteMessage.lower()=="none": OldnoteMessage=""
			print ['OLD',OldnoteType,OldnoteImage,OldnoteMessage]; print ['NEW',TypeOfMessage,NewImage,NewMessage]; 
			if (not OldnoteImage==NewImage) or (not OldnoteMessage==NewMessage): 
				TempWindow=MyWindow(noteType=TypeOfMessage,noteMessage=NewMessage,noteImage=NewImage); TempWindow.doModal(); del TempWindow; 
			elif DoFromService==True: return
			else: TempWindow=MyWindow(noteType=TypeOfMessage,noteMessage=NewMessage,noteImage=NewImage); TempWindow.doModal(); del TempWindow; 


## ################################################## ##
## ################################################## ##

